lib/upload.js

```diff
@@ -86,20 +135,34 @@ Slingshot.Upload = function (directive, metaData) {
      */
 
     send: function (file, callback) {
-      if (! (file instanceof window.File) && ! (file instanceof window.Blob))
+      if (! (file instanceof window.File) && ! (file instanceof window.Blob) && !( typeof file==='string' && /^file:\/\/.*$/i.test(file)) )
```

Comment:
This checks only for Android and BB based file path. https://github.com/apache/cordova-plugin-file#file-system-layouts